File: Seccion 12 - Punteros en C++/Seccion_12_Python/11_ejercicio_11.py

class Alumno:
    nombre: str
    edad: int
    promedio: float
    
    def __init__(self, nombre, edad, promedio):
        self.nombre = nombre
        self.edad = edad
        self.promedio = promedio

alumnos = []

def pedir_datos():
    global alumnos
    
    for i in range(3):
        nombre = input('Digite su nombre: ')
        edad = int(input('Digite su edad: '))
        promedio = float(input('Digite su promedio: '))
        
        alumno = Alumno(nombre, edad, promedio)
        alumnos.append(alumno)
        print()

def mostrar_mejor_promedio(alumnos: list):
    mayor: float = 0.0
    pos: int = 0
    
    for alumno in alumnos:
        if alumno.promedio > mayor:
            mayor = alumno.promedio
            pos = alumnos.index(alumno)
    
    print('\nEl alumno con mejor promedio es:')
    print('Nombre: {}'.format(alumnos[pos].nombre))
    print('Edad: {}'.format(alumnos[pos].edad))
    print('Promedio: {}'.format(alumnos[pos].promedio))

def main():
    
    pedir_datos()
    
    mostrar_mejor_promedio(alumnos)

if __name__ == '__main__':
    main()

File: Seccion 12 - Punteros en C++/Seccion_12_Dart/11_ejercicio_11.dart

import 'dart:io';

class Alumno {
  String nombre;
  int edad;
  double promedio;

  Alumno(this.nombre, this.edad, this.promedio);
}

List<Alumno> alumnos = new List<Alumno>(3);

void pedirDatos() {
  String nombre;
  int edad;
  double promedio;

  for (int i = 0; i < 3; i++) {
    stdout.write('Digite su nombre: ');
    nombre = stdin.readLineSync();
    stdout.write('Digite su edad: ');
    edad = int.parse(stdin.readLineSync());
    stdout.write('Digite su promedio: ');
    promedio = double.parse(stdin.readLineSync());

    Alumno alumno = new Alumno(nombre, edad, promedio);
    alumnos[i] = alumno;
    print('');
  }
}

void mostrarMejorPromedio(List<Alumno> alumnos) {
  double mayor = 0.0;
  int pos = 0;

  for (int i = 0; i < 3; i++) {
    if (alumnos[i].promedio > mayor) {
      mayor = alumnos[i].promedio;
      pos = i;
    }
  }

  print('\nEl alumno con mejor promedio es:');
  print('Nombre: ${alumnos[pos].nombre}');
  print('Edad: ${alumnos[pos].edad}');
  print('Promedio: ${alumnos[pos].promedio}');
}

void main() {

  pedirDatos();

  mostrarMejorPromedio(alumnos);
  
}